backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/UserService.cs

```diff
@@ -158,4 +185,33 @@ public async Task<UserDto> MakeAdminAsync(int userId)
         UserDto userDto = _mapper.Map<UserDto>(user);
         return userDto;
     }
+
+    public async Task<long> GetUserIdByEmail(string email)
+    {
+        var user = await _context.Users.FirstOrDefaultAsync(user=>user.Email == email);
+        return user is null ? 0 : user.Id;
+
+    }
+
+    public async Task<List<UserShortInfoDto>> GetFriends()
+    {
+        var friendshipsWithUsers = _context.Friends.Include(f => f.User).Include(f => f.Requester);
+        var requestors = await friendshipsWithUsers
+            .Where(f => f.FriendshipStatus != FriendshipStatus.Rejected && f.UserId == _authService.UserId)
+            .Select(f=>f.Requester)
+            .ToListAsync();
+
+        var acceptors = await friendshipsWithUsers
+            .Where(f => f.FriendshipStatus != FriendshipStatus.Rejected && f.RequesterId == _authService.UserId)
+            .Select(f => f.User)
+            .ToListAsync();
```

Comment:
What about something like
```suggestion
        var users= await friendshipsWithUsers
            .Where(f => f.FriendshipStatus != FriendshipStatus.Rejected && f.UserId == _authService.UserId ||  f.RequesterId == _authService.UserId)
            .Select(f=>f.UserId == _authService.UserId ? f.Requester : f.User)
            .ToListAsync();

```

Reply:
You can also add status here (as tuple, for example) UserFriendshipStatus